feat(new-trace): Redesigning span details content.

Abdkhan14 opened the pull request:
Designs: [link](https://www.figma.com/design/zfReOSQiHfODLLoFWpYVdW/Trace-V3?node-id=42-1924&node-type=frame&t=WGeiXNC70KIF6zWC-0)

Feature Flag: trace-view-new-ui

To summarize the changes, the current ui components are converted to `LegacyComponent` and the new code exists under `Component`. It'll make it easier to cleanup once we've reached general availability.

<img width="1278" alt="Screenshot 2024-11-27 at 11 50 15 AM" src="https://github.com/user-attachments/assets/03ca58ee-f60a-4bdd-b3dc-211d7d2378ad">

---------------

<img width="1311" alt="Screenshot 2024-11-27 at 11 50 46 AM" src="https://github.com/user-attachments/assets/80b9a032-7689-499c-b7f2-ad409f5d561b">

---------------

<img width="1303" alt="Screenshot 2024-11-27 at 11 52 20 AM" src="https://github.com/user-attachments/assets/33a9791f-249d-4087-a3f9-505de03cb232">

Commit:
feat(new-trace): Redesigning span details content.

Commit:
feat(new-trace): Minor iterations

Commit:
Merge branch 'master' into abdk/span-drawer-changes

Commit:
feat(new-trace): Minor iteration

Commit:
feat(new-trace): Minor iteration

Commit:
fix(new-trace): Minor iteration

Abdkhan14 committed:
feat(new-trace): Redesigning error drawer content (#81379)

Designs:
[link](https://www.figma.com/design/zfReOSQiHfODLLoFWpYVdW/Trace-V3?node-id=42-1924&node-type=frame&t=WGeiXNC70KIF6zWC-0)

Feature Flag: trace-view-new-ui

To summarize the changes, the current ui components are converted to
`LegacyComponent` and the new code exists under `Component`. It'll make
it easier to cleanup once we've reached general availability.

<img width="1308" alt="Screenshot 2024-11-27 at 2 03 34 PM"
src="https://github.com/user-attachments/assets/b360d4fe-2b60-49df-a221-f00c4e490be1">

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Abdkhan14 committed:
feat(new-trace): Redesigning autogroup drawer content. (#81414)

Designs:
[link](https://www.figma.com/design/zfReOSQiHfODLLoFWpYVdW/Trace-V3?node-id=42-1924&node-type=frame&t=WGeiXNC70KIF6zWC-0)

Feature Flag: trace-view-new-ui

<img width="1290" alt="Screenshot 2024-11-27 at 7 41 05 PM"
src="https://github.com/user-attachments/assets/88c168e4-7723-4585-a569-2f6f39ad95d9">

---------

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

0Calories commented on static/app/views/performance/newTraceDetails/traceDrawer/details/span/components/spanSummaryLink.tsx:
```diff
@@ -0,0 +1,102 @@
+
+function SpanSummaryLink(props: Props) {
+  const location = useLocation();
+  const resourceBaseUrl = useModuleURL(ModuleName.RESOURCE);
+  const queryBaseUrl = useModuleURL(ModuleName.DB);
```
These links appear to not work in some cases:

- Navigating from the Frontend Performance view > viewing a trace > clicking a DB span in the trace and clicking `View Query Summary`. I believe this is happening because `useModuleURL` does not include the `frontend` or `backend` slug. [Example](https://sentry.sentry.io/insights/frontend/trace/00bc62d08dc8430380778def6e8b4540/?fov=0%2C504&node=span-b82e3484fe1d7232&node=txn-2036dff2449e402c8d6913e41f7576f3&project=11276&query=%28%20transaction.op%3Apageload%20OR%20transaction.op%3Anavigation%20OR%20transaction.op%3Aui.render%20OR%20transaction.op%3Ainteraction%20%29&source=performance_transaction_summary&statsPeriod=24h&tab=events&timestamp=1732912723&transaction=%2Fsettings%2Forganization%2F)
- Same thing, but for asset summary when viewing a trace that you navigated to from the Backend view

But these problems are not directly related to this PR and can be fixed separately

Abdkhan14 replied:
@DominikB2014 iirc you had some PR out to fix this. Just making sure you are aware of this.

0Calories reviewed (approved):
Great stuff!

Abdkhan14 committed:
feat(new-trace): Redesigning no instrumentation drawer content. (#81432)

Designs:
[link](https://www.figma.com/design/zfReOSQiHfODLLoFWpYVdW/Trace-V3?node-id=42-1924&node-type=frame&t=WGeiXNC70KIF6zWC-0)

Feature Flag: trace-view-new-ui

To summarize the changes, the current ui components are converted to
`LegacyComponent` and the new code exists under `Component`. It'll make
it easier to cleanup once we've reached general availability.

<img width="1304" alt="Screenshot 2024-11-28 at 1 32 28 PM"
src="https://github.com/user-attachments/assets/f1882758-b6b2-488a-9354-4b55878a570c">

------------

<img width="1288" alt="Screenshot 2024-11-28 at 1 33 34 PM"
src="https://github.com/user-attachments/assets/a9bd2550-6a6e-41e2-977e-16aecb397893">

Co-authored-by: Abdullah Khan <abdullahkhan@PG9Y57YDXQ.local>

Commit:
feat(new-trace) Fixing cta to turn autogrouping off.
